### ISSUE TITLE
Clang 17.0.6 and Clang 18.1.1 crash

### ISSUE THREAD

**Issue by xyzzy12332** [llvm/llvm-project#85062]
Source code to reproduce:

```
#include <iostream>

enum MyEnum { val_A, val_B, val_C, val_D, val_E };

MyEnum  GetEnum (int i) { return i%2 ? val_A : val_E; }

int  main () {
  switch (GetEnum (0)) {
    default:
      break;

    case val_A:
    case val_B:
    case val_D:
      return 1;

    case val_C:
    case val_E:
      std::cout << "xyz" << std::endl;
      break;
  }
}

```

Can be reproduced here: [wandbox](https://wandbox.org/permlink/dYzozeVaEZVp4ZQo)
(older version can process and run the code, e.g. Clang 15.0.7)

**Comment by dtcxzyw** [comment]
Sorry, I cannot reproduce the crash with the trunk version of clang: https://godbolt.org/z/ox11G1zf7.

**Comment by xyzzy12332** [comment]
Thanks for testing. We use the precompiled versions from https://github.com/llvm/llvm-project/tags, so I guess we will have to wait for the release of LLVM 19?

**Comment by dtcxzyw** [comment]
cc @tstellar

**Comment by EugeneZelenko** [comment]
Code is also compiled with 18.1 and 17 on GodBolt: https://godbolt.org/z/Phcb7K6c6, https://godbolt.org/z/8sWfvK6qT

**Comment by xyzzy12332** [comment]
Issue seems to be resolved in 20.0.0.
